web/src/channel.ts

```diff
@@ -117,7 +132,7 @@ function call(args) {
                 reload_state.csrf_failed_handler();
             }
         }
-        orig_error(xhr, error_type, xhn);
+        orig_error!(xhr, error_type, xhn);
```

Comment:
We can fix these scoping issues without the assertions:
```diff
diff --git a/web/src/channel.ts b/web/src/channel.ts
index 246aa4dd9e..c51d3ba3bd 100644
--- a/web/src/channel.ts
+++ b/web/src/channel.ts
@@ -75,7 +75,7 @@ function call(args: AjaxRequestHandlerOptions): JQuery.jqXHR<unknown> | undefine
     if (orig_error === undefined) {
         orig_error = function () {};
     }
-    args.error = function wrapped_error(xhr, error_type, xhn) {
+    args.error = ((orig_error) => function wrapped_error(xhr, error_type, xhn) {
         if (span !== undefined) {
             span.setHttpStatus(xhr.status);
             span.finish();
@@ -132,13 +132,13 @@ function call(args: AjaxRequestHandlerOptions): JQuery.jqXHR<unknown> | undefine
                 reload_state.csrf_failed_handler();
             }
         }
-        orig_error!(xhr, error_type, xhn);
-    };
+        orig_error(xhr, error_type, xhn);
+    })(orig_error);
     let orig_success = args.success;
     if (orig_success === undefined) {
         orig_success = function () {};
     }
-    args.success = function wrapped_success(data, textStatus, jqXHR) {
+    args.success = ((orig_success) => function wrapped_success(data, textStatus, jqXHR) {
         if (span !== undefined) {
             span.setHttpStatus(jqXHR.status);
             span.finish();
@@ -152,8 +152,8 @@ function call(args: AjaxRequestHandlerOptions): JQuery.jqXHR<unknown> | undefine
             return;
         }
 
-        orig_success!(data, textStatus, jqXHR);
-    };
+        orig_success(data, textStatus, jqXHR);
+    })(orig_success);
 
     try {
         const scope = Sentry.getCurrentHub().pushScope();
```

Reply:
Right, we can do this to avoid non-null assertion here.

But I found that we can simply declare `orig_success` and `orig_error` in this way to avoid undefined values - 
```
    const orig_success = args.success ?? (() => {});
```

So did this instead!